### PR TITLE
ENH: Adding logging for emprirical measurement

### PR DESCRIPTION
**Pull request by heaven00**

### CODE REVIEW

File: pdf_to_csv.py [hunk 1]
```diff
@@ -32,7 +32,8 @@ def __init__(self):
 
     def generate_csv_file(self, input_pdf_filepath, out_csv_filepath,
                           is_header=True, identify_columns=False,
-                          temp_file_postfix="", check_page_rotation=False):
+                          temp_file_postfix="", check_page_rotation=False,
+                          measure_file=None):
```
Comment (gggodhwani) [inline pdf_to_csv.py]: log_perf or monitor? Rather than measure_file, slightly verbose name would be more helpful?

File: pdf_to_csv.py [hunk 2]
```diff
@@ -70,16 +74,32 @@ def generate_csv_file(self, input_pdf_filepath, out_csv_filepath,
...
+        '''Initialize a file object and add headers.
+
+        Args:
+            - measure_file_name (string): name/path of the file in string
+                format.
+
+        Returns:
+            The file object.
+        '''
+        measure_file_obj = open(measure_file_name, 'w')
```
Comment (gggodhwani) [inline pdf_to_csv.py]: Why pass another file, can be included in the default logs with just a separate parameter which easy to grep? "MONITORING LOG" or "PERF LOG"

File: pdf_to_csv.py [hunk 3]
```diff
@@ -156,13 +180,47 @@ def generate_page_table_data(self, input_pdf_filepath, input_pdf_obj,
             logger.warning(warning_message.format(page_num, input_pdf_filepath))
         return page_table_data
 
+    def log_data(self, input_pdf_filepath, page_num, measure_file_obj,
```
Comment (gggodhwani) [inline pdf_to_csv.py]: log_monitoring_data or log_performance_data, if possible let's keep vocab consistent throughout the file?

File: pdf_to_csv.py [hunk 4]
```diff
@@ -156,13 +180,47 @@ def generate_page_table_data(self, input_pdf_filepath, input_pdf_obj,
...
+        '''
+        if measure_file_obj is None:
+            return
+        # The file is going to be a csv thus we are going to generate a string
+        # with the following information separated by `,`
+        # filename, page_num, table detected, column_coordinates,column count
+        # and write it to the file and add a newline.
+        filename = os.path.split(input_pdf_filepath)[-1]
+        table_detected = len(column_coordinates) > 1
+        parsed_column_coordinates = '|'.join([str(num) for num in
```
Comment (gggodhwani) [inline pdf_to_csv.py]: Rationale for this?

Comment (heaven00) [inline pdf_to_csv.py]: generating a csv. | separated column coordinates would be easier to parse

File: pdf_to_csv.py (hunk 3 above)
Comment (heaven00) [inline pdf_to_csv.py]: performance monitoring

File: pdf_to_csv.py (hunk 2 above)
Comment (heaven00) [inline pdf_to_csv.py]: a clean file generation without overhead, but let me see It should be possible through the logging api too

File: pdf_to_csv.py (hunk 1 above)
Comment (heaven00) [inline pdf_to_csv.py]: yeah, I wasn't able to come up with one at that point of time

File: pdf_to_csv.py (hunk 4 above)
Comment (gggodhwani) [inline pdf_to_csv.py]: Cool then let's define it as constant in the beginning MONITORING_LOG_SEPARATOR = "|" or PERF_LOG_SEPARATOR = "|" and use it as defacto for all the monitoring/performance logs?

Comment (heaven00) [inline pdf_to_csv.py]: its specific to only one particular column. The actual separator is still  `,`

**Comment by heaven00** [comment]
@gggodhwani added it as part of logging.

**Comment by heaven00** [comment]
will reopen in sometime

**Pull request closed without merging**